src/main/java/com/aws/greengrass/security/provider/pkcs11/PKCS11CryptoKeyService.java

```diff
@@ -439,6 +444,22 @@ public String supportedKeyType() {
         return Pkcs11URI.PKCS11_SCHEME;
     }
 
+    @Override
+    public List<X509Certificate> getCertificateChain(URI privateKeyUri, URI certificateUri)
+            throws ServiceUnavailableException, KeyLoadingException {
+        KeyManager[] km = getKeyManagers(privateKeyUri, certificateUri);
+        if (km.length != 1 || !(km[0] instanceof X509KeyManager)) {
```

Comment:
Same comment as this one https://github.com/aws-greengrass/aws-greengrass-nucleus/pull/1260/files#r936824582